### PR TITLE
ddl: add M>=D checking for decimal/float/double column definition.

### PR DESCRIPTION
**Pull request by lysu**
## What have you changed? (mandatory)

For float(M,D), double(M,D) or decimal(M,D), M must be >= D (column 'a') and we need check it before create table.

## What are the type of the changes (mandatory)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (mandatory)?

- unint test

## Add a few positive/negative examples (optional)

expect:

```
mysql> create table ttt(a decimal(1, 2));
ERROR 1427 (42000): For float(M,D), double(M,D) or decimal(M,D), M must be >= D (column 'a').
```

but got:

```
Query OK, 0 rows affected (0.03 sec)
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -546,6 +546,19 @@ func checkAddColumnTooManyColumns(oldCols []*model.ColumnInfo) error {
 	return nil
 }
 
+// checkPointTypeColumnDigitLength checks
+func checkPointTypeColumnDigitLength(colDefs []*ast.ColumnDef) error {
+	for _, colDef := range colDefs {
+		if colDef.Tp.Tp != mysql.TypeNewDecimal && colDef.Tp.Tp != mysql.TypeFloat && colDef.Tp.Tp != mysql.TypeDouble {
```
Comment (coocood) [inline ddl/ddl_api.go]: `switch colDef.Tp.Tp` is cleaner.

**Comment by coocood** [comment]
LGTM

File: ddl/ddl_db_test.go [hunk 2]
```diff
@@ -1560,6 +1560,16 @@ func (s *testDBSuite) TestCreateTableWithPartition(c *C) {
 	c.Assert(part.Definitions[2].Name, Equals, "p2")
 }
 
+func (s *testDBSuite) TestCreateTableWithFloatType(c *C) {
+	s.tk.MustExec("use test")
+	s.tk.MustExec("drop table if exists t")
+	s.testErrorCode(c, "create table t (a decimal(1, 2))", tmysql.ErrMBiggerThanD)
+	s.testErrorCode(c, "create table t (a float(1, 2))", tmysql.ErrMBiggerThanD)
+	s.testErrorCode(c, "create table t (a double(1, 2))", tmysql.ErrMBiggerThanD)
+	s.mustExec(c, "create table t (a double(1, 1))")
```
Comment (zimulala) [inline ddl/ddl_db_test.go]: Could we fix the case of "alter table t add column x decimal(1, 2);" ？

Comment (lysu) [reply ddl/ddl_db_test.go]: Done, Good advise, we also check it in `change` and `modify` too~

**Pull request merged**